segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java

```diff
@@ -123,17 +151,14 @@ public PravegaConnectionListener(boolean ssl, String host, int port, StreamSegme
     //endregion
 
     public void startListening() {
-        // Configure SSL.
-        final SslContext sslCtx;
-        if (ssl) {
-            try {
-                sslCtx = SslContextBuilder.forServer(new File(this.certFile), new File(this.keyFile)).build();
-            } catch (SSLException e) {
-                throw new RuntimeException(e);
-            }
+
+        final AtomicReference<SslContext> sslCtx;
```

Comment:
This need not be AtomicReference. You are only setting its value once so there's no need to wrap it in order to access it in a lambda below.

Reply:
The SSL Context gets reloaded via the callback invoked by file modification event watcher or polling monitor. Depending on how the certificate is modified, there may be multiple occurrences of SSL context reloads.